api/api.go

```diff
@@ -0,0 +1,89 @@
+package api
+
+import (
+	"errors"
+
+	"github.com/unifiedpush/go_dbus_connector/dbus"
+	"github.com/unifiedpush/go_dbus_connector/store"
+)
+
+var client *dbus.Client
+var dataStore *store.Storage
+
+func Initialize(name string) {
+	if client != nil {
+		client.Close()
+	}
+	client = dbus.NewClient(name)
+	err := client.InitializeDefaultConnection()
+	if err != nil {
+		//TODO
+	}
+
+}
+
+//GetDistributor returns current selected distributor or empty string
+func GetDistributor() string {
+	return dataStore.Distributor
+}
+
+//GetDistributors lists all distributors that are available to register with
+// (note the difference from GetDistributor singular)
+func GetDistributors() ([]string, error) {
+	return client.ListDistributors()
+}
+
+//GetInstance returns instance from token (for internal use) or empty string if not found
+func GetInstance(token string) string {
+	for i, j := range dataStore.Instances {
+		if token == j.Token.String() {
+			return i
+		}
+	}
+
+	return ""
+}
+
+//SaveDistributor saves the distributor preference to use for future registrations
+// valid values are picked from the list returned by GetDistributors
+// all instances (registered to the previous distributor) have to unregister before running this
+func SaveDistributor(id string) error {
+	if err := storeIsEmpty(); err != nil {
+		return err
+	}
+
+	//TODO @S1m should I force this check of ensuring input is a valid distrib
```

Comment:
@p1gp1g

Reply:
An invalid distrib will just finish with notifications not working, so it is not important. But it is run rarely so we can do not optimized things. So, on my sight, we can check but don't need.